### PR TITLE
migrate to yarn v3

### PR DESCRIPTION
**Pull request by legobeat**
- Change package manager from yarnv1 to yarnv3
- devDeps: Add and enable `@lavamoat/allow-scripts`
- ci(circleci): disable HAR logs collection

#### Blocking
- #48

### CODE REVIEW

**Review by MajorLift — DISMISSED** [review]

File: package.json [hunk 1]
```diff
@@ -52,5 +54,13 @@
...
-  }
+  },
+  "lavamoat": {
+    "allowScripts": {
+      "@lavamoat/preinstall-always-fail": false,
+      "ethereumjs-util>keccak": false,
+      "ethereumjs-util>secp256k1": false
+    }
+  },
+  "packageManager": "yarn@3.6.3"
```
Comment (MajorLift) [inline package.json:65]: Using `yarn@3.3.0` here to align with core would make migration easier, unless there are updates in the latest version that we want to introduce?

Comment (Mrtenz) [reply package.json:65]: It should be trivial for `core` to update to a more recent Yarn version too.

File: .yarnrc.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+logFilters:
+  - code: YN0004
+    level: discard
+
+nodeLinker: node-modules
+
+plugins:
+  - path: .yarn/plugins/@yarnpkg/plugin-allow-scripts.cjs
+    spec: 'https://raw.githubusercontent.com/LavaMoat/LavaMoat/main/packages/yarn-plugin-allow-scripts/bundles/@yarnpkg/plugin-allow-scripts.js'
+  - path: .yarn/plugins/@yarnpkg/plugin-constraints.cjs
```
Comment (Mrtenz) [inline .yarnrc.yml:14]: I'd either copy the constraints from the module template as well, or remove this plugin.

File: .github/workflows/build-lint-test.yml [hunk 3]
```diff
@@ -32,7 +32,7 @@ jobs:
         with:
           node-version: ${{ matrix.node-version }}
           cache: 'yarn'
-      - run: yarn --immutable --immutable-cache --ignore-engines
+      - run: yarn --immutable
```
Comment (Mrtenz) [inline .github/workflows/build-lint-test.yml:35]: In the module template we use `--immutable-cache` for these steps.

**Review by Mrtenz — CHANGES_REQUESTED** [review]

**Review by Mrtenz — APPROVED** [review]

**Pull request merged**